### PR TITLE
[Task #1141] Add Debian release and digest to Ruby image reference

### PR DESCRIPTION
**Pull request by vr4b4c**
## Summary

Add Debian release and digest to Ruby image reference. For patch-level versions in 3.4, trixie images is built only for Ruby version 3.4.5, hence update for `.ruby-version`.

### libyaml
To compile native extension for `psych`, libyaml headers need to be present on the build system.

**Related issue**: [#1141](https://app.productive.io/1-infinum/projects/2274/tasks/task/13471789?board=268199&filter=MjA2MTY2)

### Type

- [x] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

**Review by nikajukic — APPROVED** [review]

**Review by djordjelacmanovic — APPROVED** [review]

File: templates/Dockerfile.tt [hunk 1]
```diff
@@ -1,10 +1,11 @@
 ARG RUBY_VERSION=<%= RUBY_VERSION %>
-FROM ruby:$RUBY_VERSION-slim AS base
+FROM ruby:$RUBY_VERSION-slim-trixie@sha256:0d2adfa1930d67ee79e5d16c3610f4fbed43c98e98dbda14c2811b8197211c74 AS base
```
Comment (lovro-bikic) [inline templates/Dockerfile.tt:2]: What was the decision against using the image without the release name?
```suggestion
FROM ruby:$RUBY_VERSION-slim@sha256:0d2adfa1930d67ee79e5d16c3610f4fbed43c98e98dbda14c2811b8197211c74 AS base
```

I know that we're subscribed to Debian releases now so we'll know when to update the image, but I think even this suggestion works better because you're always on the same release and version, and Dependabot will notify you when there's a new release.

Comment (vr4b4c) [reply templates/Dockerfile.tt:2]: I'm OK with both approaches. This is more conservative (stable) approach and less likely for contributor to unintentionally break the production. Official [docs](https://hub.docker.com/_/ruby#image-variants) also suggest using this approach
> If your image needs to install any additional packages beyond what comes with the image, you'll likely want to specify one of these explicitly to minimize breakage when there are new releases of Debian.

In any case, this decision is easy to convert at any point.

Comment (lovro-bikic) [reply templates/Dockerfile.tt:2]: Good point, I agree with keeping the release name in the tag then 👍
